user/src/com/google/gwt/user/client/rpc/core/java/util/LinkedHashMap_CustomFieldSerializer.java

```diff
@@ -105,15 +108,30 @@ private static boolean getAccessOrder(LinkedHashMap instance) {
     if (!reflectionHasFailed.get()) {
       try {
         Field f = accessOrderField.get();
-        if (f == null || !f.isAccessible()) {
+        boolean isAccessible = f != null && f.isAccessible();
+        if (!isAccessible) {
           f = LinkedHashMap.class.getDeclaredField("accessOrder");
           synchronized (f) {
-            // Ensure all threads can see the accessibility.
-            f.setAccessible(true);
+            try {
+              // see if we can *try* setting the accessOrder field accessible:
+              final Method trySetAccessible = AccessibleObject.class.getDeclaredMethod("trySetAccessible");
+              // no exception, then we are on Java 9 or beyond
+              if ((boolean) trySetAccessible.invoke(f)) {
+                accessOrderField.set(f);
+                isAccessible = true;
+              }
+            } catch (NoSuchMethodException e) {
+              // Java <= 8: it won't create a warning but it may fail with an exception
+              // Ensure all threads can see the accessibility.
+              f.setAccessible(true);
+              accessOrderField.set(f);
+              isAccessible = true;
+            }
           }
-          accessOrderField.set(f);
         }
-        return ((Boolean) f.get(instance)).booleanValue();
+        if (isAccessible) {
```

Comment:
To confirm (but not necessarily objecting to this change), this `if` isn't strictly required, since `Field.get()` will throw an IllegalAccessException, which will fall through and still land on the `reflectionHasFailed.set(true)` below, right?

If so, we only need the trySetAccessible/setAccessible change above, and the additional catch block

Reply:
The "if" is supposedly faster than letting the code run into an exception that needs to be caught. That's why I added it.